connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -490,3 +498,48 @@ export async function getFoldersToSync(connectorId: ModelId) {
 
   return foldersIds;
 }
+
+export async function garbageCollector(
+  connectorId: ModelId,
+  startTs: number
+): Promise<number> {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const authCredentials = await getAuthObject(connector.connectionId);
+  const files = await GoogleDriveFiles.findAll({
+    where: {
+      connectorId: connectorId,
+      garbageCollectedAt: { [Op.or]: [{ [Op.lt]: new Date(startTs) }, null] },
+    },
+    limit: 100,
```

Comment:
What if there are more than 100 here?

Reply:
is it possible for garbageCollectedAt to be not less than startTs or null?

Reply:
From the code here I don't understand how that would be possible, so these are ~all the files no? hence there will always be more than 100?

I'm probably missing something :grimacing:

Reply:
Ah I see the outer loop in the workflow `googleDriveGarbageCollectorWorkflow` 

Probably worth a comment for future readers

Reply:
Sure I will add it but a lot of things are processed in batches  in connectors. It's to have a retry at a reasonable cost and still be somewhat efficient.

Reply:
We use garbageCollectedAt as a pagination mechanism.
Here is what is going on:
- Get all files synced from the connectors DB, page by page (of 100).
- We check if this file should still be synchronized by calling `isObjectInFolder(objectId, selectedFolders)`
- if we keep it, just update the garbageCollectedAt to now()
- if not, we just delete it remotely and locally.